src/scheduler.rs

```diff
@@ -322,9 +325,16 @@ async fn smtp_loop(ctx: Context, started: Sender<()>, smtp_handlers: SmtpConnect
                     interrupt_info = Default::default();
                 }
                 None => {
-                    if let Err(err) = send_smtp_messages(&ctx, &mut connection).await {
+                    let res = send_smtp_messages(&ctx, &mut connection).await;
+                    if let Err(err) = &res {
                         warn!(ctx, "send_smtp_messages failed: {:#}", err);
                     }
+                    let success = res.unwrap_or_default();
+                    if success {
+                        timeout = MIN_TIMEOUT;
+                    } else {
+                        timeout = timeout.saturating_mul(2);
```

Comment:
There are 6 SMTP retries, 2^6=64, we are starting at 0.5min, so the longest timeout is 32 minutes if my calculations are correct. This _should_ be fine since `send_msg_to_smtp()` doesn't increase the retry counter if `connect_configured()` fails, but still I think it would be better to multiply with 3 here to be safe in case `connect_configured()` returns Ok for some reason, or a message fails due to some other temporary error:
```suggestion
                        timeout = timeout.saturating_mul(3);
```
This way, the longest timeout is about 15 days (3^6=729, 729/2/24=15). And also, this might save some battery since the device is awoken less often.

Reply:
Updated to use 30 seconds and multiply by 3